### DISCUSSION TITLE
How do I set Exception Breakpoints in Theia UI?

### DISCUSSION THREAD

**Discussion by tsiyona** [eclipse-theia/theia#8006] [General]
I am writing a debug-adapter for Java, and would like to set exception breakpoints in Theia UI.

I am setting filters in the debugger capabilities returned in 'initialize' DAP response as follows:
	

	`export class ExceptionBreakpointFilter {
		label: string;
		filter: string;

		public constructor(value: string, label: string) {
			this.filter = value;
			this.label = label;
		}
	 }
         protected initializeRequest(response: DebugProtocol.InitializeResponse,
				args: DebugProtocol.InitializeRequestArguments): void {

		response.body = response.body || {};

		// set some capabilities . . .
		
                // now this enables the setExceptionBreakpoints 
		let exceptionFilters: Array<ExceptionBreakpointFilter>  = Array<ExceptionBreakpointFilter>();
		exceptionFilters.push(new ExceptionBreakpointFilter("uncaught", "Uncaught Exceptions"));
		exceptionFilters.push(new ExceptionBreakpointFilter("caught", "Caught Exceptions"));
		response.body.exceptionBreakpointFilters = exceptionFilters;

		response.body.supportsExceptionInfoRequest = true;
		response.body.supportsExceptionOptions = true;
		this.sendResponse(response);
	}
`
but I cannot see how this enables setting exception breakpoints in the UI.

Is this capability supported by Theia?
If yes, can someone please assist?
Thank you!

**Comment by akosyakov** [comment]
You should start the debug session at least once after that you should be able to set exception breakpoints in the breakpoints view. The issue is that we don't know whether installed debug adapters support exception breakpoints without running them at least once.

**Reply by tsiyona** [reply]
I am not sure I follow. I started a debug session many times after I set the exception filters. The breakpoints view does not show anything that indicates I can set an exception breakpoint. It only shows the standard options.

![breakpoints-view](https://user-images.githubusercontent.com/9335329/84677671-b9978b00-af37-11ea-9651-8fff8af22069.png)

What additional option should I have had there after Theia learns that my debug adapter supports exception breakpoints?

**Reply by akosyakov** [reply]
ok, it sounds like a bug

**Reply by akosyakov** [reply]
Could you file an issue with steps to reproduce, i.e. how to run your DAP adapter.

**Reply by tsiyona** [reply]
I opened the following issue: [#8019](https://github.com/eclipse-theia/theia/issues/8019)
